### PR TITLE
refactor: resultError always emitted on errors

### PR DESCRIPTION
**Pull request by luddd3**
The check of self.listeners('resultError') is unnecessary because it will always result in an array which evaluates to true.

### CODE REVIEW

**Comment by jsumners** [comment]
@luddd3 then wouldn't the more appropriate change be to check if that array's length is `> 0`?

**Comment by jsumners** [comment]
Also, are you able to cover this change with a unit test?

File: lib/client/client.js [hunk 1]
```diff
@@ -1150,7 +1150,7 @@ Client.prototype._sendSocket = function _sendSocket (message,
   var sentEmitter = false
 
   function sendResult (event, obj) {
-    if (event === 'error' && self.listeners('resultError')) {
+    if (event === 'error') {
```
Comment (UziTech) [inline lib/client/client.js]:
```suggestion
    if (event === 'error' && self.listenerCount('resultError')) {
```

**Comment by luddd3** [comment]
> Also, are you able to cover this change with a unit test?

This is already covered by a unit test. See `client.test.js` line 1438, "resultError handling".

File: lib/client/client.js (hunk 1 above)
Comment (luddd3) [reply lib/client/client.js]: Seems unnecessary. If there are no listeners, the emit is effectually a noop. It is not custom to look for listeners before emitting an event and it is not done anywhere else in this code base.

Comment (jsumners) [reply lib/client/client.js]: Makes sense. I was only going on the intent implied by the original code in this change set.

**Comment by jsumners** [comment]
@luddd3 can you do a force push an empty commit or something? GitHub Actions seems to be stuck and will not cancel (I have tried multiple times).

**Comment by luddd3** [comment]
@jsumners I think something is wrong with the github action :(

**Comment by jsumners** [comment]
I'll have to debug it this weekend :(

**Comment by jsumners** [comment]
The CI issue was https://github.com/tapjs/node-tap/issues/620

Thank you for contributing. This has been released as `2.0.0-pre.4` and installable as `ldapjs@next`.

**Comment by luddd3** [comment]
Thank you!

**Comment by jsumners** [comment]
⚠️ This issue has been locked due to age. If you have encountered a recent
problem that seems to be covered by this issue, please open a new issue.

Please include a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)
when opening a new issue.

**Pull request merged**